### PR TITLE
Reduce basic install scope

### PR DESCRIPTION
**Pull request by aashah**
* **Separate out an `install-core.sh` from `install.sh`** The old `install` did too much and was too opinionated. `install-core` is now mostly about installing binaries and core utilities. It does not install anything that improves developer efficiency, or have opinions on things like shells and editors.
* **Introduced `lib` folder** The intent behind this folder is similar to `scripts`, a folder that can be `source` or added to the `PATH`. The configuration for it however, is not within there. Instead, to not break existing folks, I load them in via a bash-it plugin (`bash-it/load-capi-workspace-lib.bash`).
  * ❗❗ I don't know how bash-it handles renaming a plugin's filename ❗❗

I think the best way of evaluating this PR is by checking out the branch locally.

### CODE REVIEW

File: Brewfile-core [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+tap 'k14s/tap'
+tap 'pivotal/tap'
+tap 'buildpack/tap'
+
+# Core utilities
+brew 'git'
+brew 'lastpass-cli'
+brew 'wget'
+brew 'coreutils'
+brew 'direnv'
```
Comment (cwlbraa) [inline Brewfile-core:11]: bash and bash-completion should be in here. no one should have to care about whatever random ancient version of bash OSX comes with, and we shouldn't make people who can't install the monolithic thing to be typing every letter of every command on the command line.

File: lib/misc.bash [hunk 2]
```diff
@@ -1,16 +1,9 @@
-#FASD
-alias v='fasd -e vim'
-[ -f /usr/local/etc/profile.d/autojump.sh ] && . /usr/local/etc/profile.d/autojump.sh
-
-alias b='bundle exec'
-alias bake='echo "bundling..." && bundle install --quiet && echo "done bundling" && bundle exec rake'
```
[the remark this answers is not in the document]
Comment (aashah) [reply lib/misc.bash:6]: I agree. The PR may be showing it weird, but they've been moved to https://github.com/cloudfoundry/capi-workspace/blob/reduce-basic-install-scope/lib/aliases.bash

File: lib/misc.bash [hunk 3]
```diff
@@ -1,16 +1,9 @@
...
-alias pgbake='echo "bundling..." && bundle install --quiet && echo "done bundling" && DB=postgres bundle exec rake'
-alias mybake='echo "bundling..." && bundle install --quiet && echo "done bundling" && DB=mysql bundle exec rake'
```
[the remark this answers is not in the document]
Comment (aashah) [reply lib/misc.bash:8]: the PR may be showing it weird, but they've been moved to https://github.com/cloudfoundry/capi-workspace/blob/reduce-basic-install-scope/lib/aliases.bash

File: Brewfile-core (hunk 1 above)
Comment (aashah) [reply Brewfile-core:11]: OS X no longer deploys bash as the default shell.

I thought about it, and it helps to read this comment. I'll add it back, if only for the fact that on _some_ peoples machines, are very old and insecure versions of bash.

**Comment by aashah** [comment]
Sharing a data point here -- Piyali got a new machine and ran `./install.sh`. It went fairly far, here are some notes:

* we need to address cf7 going GA outside this PR. tiny failure here.
* it failed somewhere inside the `git-hooks.sh`.

**Comment by aashah** [comment]
Some notes from running through install with Chris Selzo:

* he ran into similar issues as Piyali (pip & iterm)
* we took a closer look at item:
  * removing bash_it => no crash
  * removing `load-capi-workspace-lib.bash` => no crash
  * one-by-one, finding the culprit inside `load-capi-workspace-lib.bash`...removing `environment-variables.bash` => no crash
  * weirdly...bringing environment-variables back, and removing it again => **crash**

basically...couldn't reproduce it properly.

**Comment by aashah** [comment]
Did a complete `./install.sh` on my machine (didn't have issues with pip, cause I did that manually && separately).

Could not reproduce the issues they were having with iterm.

**Comment by aashah** [comment]
Magic stuff happened on Chris's workstation that we can't explain...but it's working now.

**Comment by aashah** [comment]
Same thing happened with Piyali, she was running on the commit prior to cd362d3, so removing `load-capi-workspace.lib.bash` "fixed" the issue for her as well...but when: bringing it back && changing nothing about capi-workspace/lib...everything started working again. How does that make any sense?

Either way, when in a working state, we just spent a hot second bringing her to the HEAD of this branch and nothing was broken. Weird stuff yo.
